### PR TITLE
Add code coverage for DataGridViewCellConverter

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

# Proposed changes
Add unit tests for DataGridViewCellConverter.cs to test its methods
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12622)

### CODE REVIEW

**Review by ricardobossan — DISMISSED** [review]
All LGTM!

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    private readonly DataGridViewCellConverter _converter = new();
+
+    [WinFormsTheory]
+    [InlineData(typeof(InstanceDescriptor), true)]
+    [InlineData(typeof(string), true)]
+    public void CanConvertTo_ReturnsExpected(Type destinationType, bool expected)
+    {
+        _converter.CanConvertTo(null, destinationType).Should().Be(expected);
+    }
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs:21]: We prefer expression-bodied syntax - 
```suggestion
    public void CanConvertTo_ReturnsExpected(Type destinationType, bool expected) =>
        _converter.CanConvertTo(null, destinationType).Should().Be(expected);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    [WinFormsFact]
+    public void ConvertTo_InstanceDescriptorWithDataGridViewCell_ReturnsInstanceDescriptor()
+    {
+        using DataGridViewTextBoxCell cell = new();
+        Type destinationType = typeof(InstanceDescriptor);
+
+        _converter.ConvertTo(null, CultureInfo.InvariantCulture, cell, destinationType).Should().NotBeNull();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs:29]: Please add the argument name before the literal - `context: null`.
This comment applies to several places in this file.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    [WinFormsFact]
+    public void ConvertTo_InstanceDescriptorWithNonDataGridViewCell_ThrowsNotSupportedException()
+    {
+        object value = new();
+        Type destinationType = typeof(InstanceDescriptor);
+
+        Action action = () => _converter.ConvertTo(null, CultureInfo.InvariantCulture, value, destinationType);
+        action.Should().Throw<NotSupportedException>()
+            .WithMessage("'DataGridViewCellConverter' is unable to convert 'System.Object' to 'System.ComponentModel.Design.Serialization.InstanceDescriptor'.");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs:41]: I assume that we don't own this resource string, that it comes from the runtime assemblies, right? If we owned it, we could reference it by `SR.<resource name>`

Comment (Zheng-Li01) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellConverterTests.cs:39]: Yes, this string comes from the runtime assemblies,

**Review by lonitra — APPROVED** [review]
LGTM

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**